ts_src/psbt.ts

```diff
@@ -584,25 +584,45 @@ export class Psbt {
       (resolve, reject): void => {
         if (!keyPair || !keyPair.publicKey)
           return reject(new Error('Need Signer to sign input'));
-        const { hash, sighashType } = getHashAndSighashType(
-          this.data.inputs,
-          inputIndex,
-          keyPair.publicKey,
-          this.__CACHE,
-          sighashTypes,
-        );
+        let result: Buffer | Promise<Buffer>;
+        let hashAndSighash: { hash: Buffer; sighashType: number };
+        let resultPromise: Promise<Buffer>;
+        try {
+          hashAndSighash = getHashAndSighashType(
+            this.data.inputs,
+            inputIndex,
+            keyPair.publicKey,
+            this.__CACHE,
+            sighashTypes,
+          );
 
-        Promise.resolve(keyPair.sign(hash)).then(signature => {
-          const partialSig = [
-            {
-              pubkey: keyPair.publicKey,
-              signature: bscript.signature.encode(signature, sighashType),
-            },
-          ];
+          result = keyPair.sign(hashAndSighash.hash);
 
-          this.data.updateInput(inputIndex, { partialSig });
-          resolve();
-        });
+          if (result && typeof (result as any).then === 'function') {
```

Comment:
Note: Apparently the ECMAScript definition of a Promise is "check for then function in the prototype, if so, treat as Promise"... so this seems like the best way to check if it is a Promise.

Reply:
`result instanceof Promise` also seems fine.